In [63]:
import dill
from sklearn.pipeline import Pipeline
import Traitement_de_texte as tt
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.base import clone
from sklearn.dummy import DummyClassifier
from sklearn.model_selection import train_test_split, GridSearchCV, StratifiedKFold
from sklearn.linear_model import LogisticRegression
from sklearn.dummy import DummyClassifier
from sklearn.pipeline import Pipeline
from Traitement_de_texte import preprocess_text
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.preprocessing import FunctionTransformer
from sklearn.compose import ColumnTransformer
from sklearn.naive_bayes import MultinomialNB

In [8]:
chemin = r'C:\Users\HP\Desktop\ISEP2\Semestre2\Machine Learning\Projet\Models\train_test_split.dill'
with open(chemin, 'rb') as f:
    x_train, x_test, y_train, y_test = dill.load(f)

In [4]:
import logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

### Chargement des modéles

In [5]:
from pathlib import Path
import dill

# Répertoires des modèles
MODEL_DIR = r'C:\Users\HP\Desktop\ISEP2\Semestre2\Machine Learning\Projet\Models'

# Chemins vers les modèles
model_paths = {
    "nb": Path(MODEL_DIR, "nb_best_model_optimized.dill"),
    "rl": Path(MODEL_DIR, "rl_best_model_optimized.dill"),
    "rf": Path(MODEL_DIR, "rf_best_model_optimized.dill"),
    "xgb": Path(MODEL_DIR, "xgb_best_model_optimized.dill")
}

# Chargement des modèles
models = {}
for name, path in model_paths.items():
    with open(path, "rb") as f:
        models[name] = dill.load(f)
        print(f"{name.upper()} chargé avec succès.")


NB chargé avec succès.
RL chargé avec succès.
RF chargé avec succès.
XGB chargé avec succès.


In [6]:
from sklearn.metrics import roc_auc_score, classification_report

for name, model in models.items():
    print(f"\n--- {name.upper()} ---")
    print(f"Seuil optimal : {model.threshold}")
    
    # Prédictions sur le jeu de test
    proba = model.predict_proba(x_test)[:, 1]
    preds = model.predict(x_test)
    
    # Évaluation
    print(f"AUC : {roc_auc_score(y_test, proba):.4f}")
    print(f"Classification Report :\n{classification_report(y_test, preds)}")



--- NB ---
Seuil optimal : 0.34
AUC : 0.8803
Classification Report :
              precision    recall  f1-score   support

           0       0.92      0.91      0.91      1851
           1       0.62      0.65      0.63       423

    accuracy                           0.86      2274
   macro avg       0.77      0.78      0.77      2274
weighted avg       0.86      0.86      0.86      2274


--- RL ---
Seuil optimal : 0.37
AUC : 0.8950
Classification Report :
              precision    recall  f1-score   support

           0       0.92      0.93      0.92      1851
           1       0.67      0.63      0.65       423

    accuracy                           0.87      2274
   macro avg       0.79      0.78      0.79      2274
weighted avg       0.87      0.87      0.87      2274


--- RF ---
Seuil optimal : 0.52
AUC : 0.8877
Classification Report :
              precision    recall  f1-score   support

           0       0.87      0.98      0.92      1851
           1       0.82    

### Best modéle

##### Le meilleur modéle obtenu est la régression logistique avec un auc de 0.8950, un seuil de 0.37 et un f1-score de 0.65 pour la classe 1

### Features importance